textacy/fileio/read.py

```diff
@@ -0,0 +1,187 @@
+            for example::
+
+                {"title": "Harrison Bergeron", "text": "The year was 2081, and everybody was finally equal."}{"title": "2BR02B", "text": "Everything was perfectly swell."}
+
+        mode (str, optional)
+        encoding (str, optional)
+        buffersize (int, optional): number of bytes to read in as a chunk
+
+    Yields:
+        dict: next valid JSON object, converted to native Python equivalent
+    """
+    with io.open(filename, mode=mode, encoding=encoding) as f:
+        buffer = ''
+        for chunk in iter(partial(f.read, buffersize), ''):
+            buffer += chunk
+            while buffer:
+                try:
+                    result, index = JSON_DECODER.raw_decode(buffer)
+                    yield result
+                    buffer = buffer[index:]
+                # not enough data to decode => read another chunk
+                except ValueError:
+                    break
+
+
+def read_file(filename, mode='rt', encoding=None):
+    """
+    Read the full contents of a file. Files compressed with gzip or bz2 are handled
+    automatically.
+    """
+    _open = gzip.open if filename.endswith('.gz') \
+        else bz2.open if filename.endswith('.bz2') \
+        else io.open
+    with _open(filename, mode=mode, encoding=encoding) as f:
+        return f.read()
+
+
+def read_file_lines(filename, mode='rt', encoding=None):
+    """
+    Read the contents of a file, line by line. Files compressed with gzip or bz2
+    are handled automatically.
+    """
+    _open = gzip.open if filename.endswith('.gz') \
+        else bz2.open if filename.endswith('.bz2') \
+        else io.open
+    with _open(filename, mode=mode, encoding=encoding) as f:
+        for line in f:
+            yield f
+
+
+def read_spacy_docs(spacy_vocab, filename):
+    """
+    Stream ``spacy.Doc`` s from disk at ``filename`` where they were serialized
+    using Spacy's ``spacy.Doc.to_bytes()`` functionality.
+
+    Args:
+        spacy_vocab (``spacy.Vocab``): the spacy vocab object used to serialize
+            the docs in ``filename``
+        filename (str): /path/to/file on disk from which spacy docs will be streamed
+
+    Yields:
+        the next deserialized ``spacy.Doc``
+    """
+    with io.open(filename, mode='rb') as f:
+        for bytes_string in Doc.read_bytes(f):
+            yield Doc(spacy_vocab).from_bytes(bytes_string)
+
+
+def get_filenames_in_dir(dirname, mode='rt', encoding='utf8',
+                         file_type=None, subdirs=False):
+    """
+    Yield the full names of files under directory ``dirname``, optionally
+    filtering for a certain file type and crawling all subdirectories.
+
+    Args:
+        dirname (str): /path/to/dir on disk where files to read are saved
+        mode (str, optional): mode with which to open files; default is probably correct
+        encoding (str, optional): encoding with which to open files
```

Comment:
Ha, good catch. I refactored the function but forgot to drop the obsolete args. Will commit in a sec.